embulk-output-jdbc/src/main/java/org/embulk/output/jdbc/PageReaderRecord.java

```diff
@@ -71,24 +84,191 @@ public Value getJson(Column column)
+            }
+
+            @Override
+            public void doubleColumn(Column column)
+            {
+                write(writer, record.getDouble(column));
+            }
+
+            @Override
+            public void stringColumn(Column column)
+            {
+                write(writer, record.getString(column));
+            }
+
+            @Override
+            public void timestampColumn(Column column)
+            {
+                write(writer, record.getTimestamp(column));
+            }
+
+            @Override
+            public void jsonColumn(Column column)
+            {
+                write(writer, record.getJson(column));
+            }
+        });
+        writer.println();
+        writer.flush();
+    }
+
+    public void foreachRecord(Function<? super Record, Boolean> function) throws IOException
+    {
+        File tmpFile = createTempFile();
+        try (CSVParser reader = openReader(readRecordsFile); CSVPrinter tmpWriter = openWriter(tmpFile)) {
+            for (CSVRecord r : reader) {
+                MemoryRecord record = new MemoryRecord(pageReader.getSchema().getColumnCount());
+                pageReader.getSchema().visitColumns(new ColumnVisitor()
+                {
+                    @Override
+                    public void booleanColumn(Column column)
+                    {
+                        setValue(record, column, r.get(column.getIndex()), Boolean.class);
+                    }
+
+                    @Override
+                    public void longColumn(Column column)
+                    {
+                        setValue(record, column, r.get(column.getIndex()), Long.class);
+                    }
+
+                    @Override
+                    public void doubleColumn(Column column)
+                    {
+                        setValue(record, column, r.get(column.getIndex()), Double.class);
+                    }
+
+                    @Override
+                    public void stringColumn(Column column)
+                    {
+                        setValue(record, column, r.get(column.getIndex()), String.class);
+                    }
+
+                    @Override
+                    public void timestampColumn(Column column)
+                    {
+                        setValue(record, column, r.get(column.getIndex()), Instant.class);
+                    }
+
+                    @Override
+                    public void jsonColumn(Column column)
+                    {
+                        setValue(record, column, r.get(column.getIndex()), Value.class);
+                    }
+                });
+                if (function.apply(record)) {
+                    writeRow(tmpWriter, record);
+                    tmpWriter.flush();
+                }
+            }
+        }
+        setReadRecords(tmpFile);
+    }
+
+    private void setValue(MemoryRecord record, Column column, String str, Class<?> obj)
+    {
+        if (str == null) {
```

Comment:
ここはミスでした。nullと"null"両方の確認が必要です。
明日に確認します。
```suggestion
        if (str == null || str.equals("null")) {
```

Reply:
不要です。`nil`は`"null"`として書き出し、`"null"`は`null`として読み込むので